readthedocs/projects/models.py

```diff
@@ -1040,6 +1044,37 @@ def get_original_stable_version(self):
         )
         return original_stable
 
+    def update_latest_version(self):
+        """
+        If the current latest version is machine created, update it.
+
+        A machine created LATEST version is an alias for the default branch/tag,
+        so we need to update it to match the type and identifier of the default branch/tag.
+        """
+        latest = self.versions.filter(slug=LATEST).first()
+        if not latest:
+            latest = self.versions.create_latest()
+        if not latest.machine:
+            return
+
+        # default_branch can be a tag or a branch name!
+        default_version_name = self.get_default_branch()
+        original_latest_type = (
+            self.versions(manager=INTERNAL)
+            .exclude(slug=LATEST)
```

Comment:
Why are we excluding the `latest` slug here? 🤔

Reply:
We want to get the original version that latest points to. But latest versions have the verbose name set to `latest`, so probably isn't needed.

Reply:
oh, this is the same cas as `project.get_stable_version` and `project.get_original_stable_version`? 😄 

It probably make sense to write those two methods for `latest` too: `.get_latest_version` and `get_original_latest_version`, right?

Reply:
> oh, this is the same cas as project.get_stable_version and project.get_original_stable_version? 😄

yeah

> It probably make sense to write those two methods for latest too: .get_latest_version and get_original_latest_version, right?

Maybe, I don't think we currently require getting the original latest outside this block.

Reply:
> Maybe, I don't think we currently require getting the original latest outside this block.

I will need it to implement https://github.com/readthedocs/addons/issues/132 as well. Since I need to know the exact branch the "original latest" points to 😄

Reply:
https://github.com/readthedocs/readthedocs.org/pull/10738/commits/a81b5ae69c47ec88fe2dee677d99577f6d0553c2